wsi/drm/drmdisplay.cpp

```diff
@@ -409,6 +409,59 @@ bool DrmDisplay::GetDisplayAttribute(uint32_t config /*config*/,
   return status;
 }
 
+uint32_t DrmDisplay::GetPreferedDisplayMode(size_t modes_size) {
+  if (modes_size > 1) {
+    SPIN_LOCK(display_lock_);
+    for (size_t i = 0; i < modes_size; i++) {
+      // There is only one preferred mode per connector.
+      if (modes_[i].type & DRM_MODE_TYPE_PREFERRED) {
+        prefer_display_mode_ = i;
+        IHOTPLUGEVENTTRACE("Preferred display config is found. index: %d", i);
+        break;
+      }
+    }
+    SPIN_UNLOCK(display_lock_);
+  }
+  return prefer_display_mode_;
+}
+
+uint32_t DrmDisplay::GetPerformaceDisplayMode(size_t modes_size,
+                                              uint32_t prefer_display_mode) {
+  perf_display_mode_ = prefer_display_mode;
```

Comment:
It is bad coding here...
since perfer_display_mode_ is a member of this class, you don't need to pass and assign it everywhere